app/models/repository/validations.rb

```diff
@@ -20,76 +18,113 @@ module Repository::Validations
                      if: :path_changed?
 
     validates_with UnreservedValidator, if: :path_changed?
+
+    validates :state, inclusion: {in: Repository::Importing::STATES}
+    validates_with SourceTypeValidator, if: :source_address?
+    validates :remote_type, presence: true, if: :source_address?
+    validates_with RemoteTypeValidator
+
+    validates_with RemoteAccessValidator
+    validates :access,
+      presence: true,
+      inclusion: {in: Repository::Access::OPTIONS}
   end
 
-  def set_path
-    self.path ||= name.parameterize if name
+  class RemoteAccessValidator < ActiveModel::Validator
+    def validate(record)
+      if record.mirror? && (record.private_rw? || record.public_rw?)
+        record.errors[:access] =
+          'Error! Write access is not allowed for a mirrored repositry.'
+      end
+    end
   end
-end
 
-class UnreservedValidator < ActiveModel::Validator
+  class UnreservedValidator < ActiveModel::Validator
+    def validate(record)
+      if reserved_name?(record.path)
```

Comment:
Style/IfUnlessModifier: Favor modifier `if` usage when having a single-line body. Another good alternative is the usage of control flow `&&`/`||`.